External/FEXCore/Source/Utils/AllocatorOverride.cpp

```diff
@@ -0,0 +1,139 @@
+#include <FEXCore/Utils/Allocator.h>
+#include <FEXCore/Utils/CompilerDefs.h>
+#include <FEXCore/Utils/LogManager.h>
+
+#include <fmt/format.h>
+
+#include <cstddef>
+#include <cstdint>
+#include <fstream>
+#include <sstream>
+#include <unistd.h>
+
+extern "C" {
+#ifdef GLIBC_ALLOCATOR_FAULT
+  // glibc allocator faulting. If anything uses a glibc allocator symbol then cause a fault.
+  // Ensures that FEX only ever allocates memory through its allocator routines.
+  // If thunks are enabled then this should crash immediately. Thunks should still go through glibc allocator.
```

Comment:
The phrasing is a bit convoluted here. Generally, it's unclear what you mean if something "should" do something (is this an expectation to the user or to the implementation? If it's the latter, "should" indicates a degree of uncertainty... why is there uncertainty and what should the user do about it?)

For this comment specifically:
* What does thunks being enabled have to do with crashing here?
* I'm not sure what information to extract from "Thunks should still go through glibc allocator". My best guess actually contradicts the preceding sentence.

Something like this might be better:
* "The majority of FEX-internal code should avoid using the glibc allocator. To ensure glibc allocations don't accidentally slip in, we override these glibc functions with faulting variants."
* "A notable exception is thunks[, which should still ...]. Other minor exceptions throughout FEX use the YesIKnowImNotSupposedToUseTheGlibcAllocator helper to temporarily disable faulting."